sdk/nodejs/tests/output.spec.ts

```diff
@@ -409,5 +440,23 @@ describe("output", () => {
             const output1 = output({ a: 1, b: 2 });
             assert.strictEqual((<any>output1).__pulumiResource, undefined);
         }));
+
+        it("lifts properties from values with nested unknowns", asyncTest(async () => {
+            const output1 = output({ foo: "foo", bar: unknown, baz: Promise.resolve(unknown) });
+
+            assert.equal(await output1.isKnown, false);
+
+            const result1 = output1.foo;
+            assert.equal(await result1.isKnown, true);
+            assert.equal(await result1.promise(), "foo");
+
+            const result2 = output1.bar;
+            assert.equal(await result2.isKnown, false);
+            assert.equal(await result2.promise(), unknown);
+
+            const result3 = output1.baz;
+            assert.equal(await result3.isKnown, false);
+            assert.equal(await result3.promise(), unknown);
```

Comment:
let's hae a fun case with something like `{ foo: [unknown, "val" }` and the code having `output.foo[0]` and `output.foo[1]`.

Reply:
Done.